packages/core/src/tw/component/components/paragraph.ts

```diff
@@ -294,13 +314,16 @@ export class ParagraphViewNode extends BlockViewNode<ParagraphLayoutNode> {
     }
 
     onLayoutDidUpdate() {
+        const style = this.layoutNode.getStyle();
         this.domContainer.style.width = `${this.layoutNode.getWidth()}px`;
         this.domContainer.style.height = `${this.layoutNode.getHeight()}px`;
         this.domContainer.style.paddingTop = `${this.layoutNode.getPaddingTop()}px`;
         this.domContainer.style.paddingBottom = `${this.layoutNode.getPaddingBottom()}px`;
         this.domContainer.style.paddingLeft = `${this.layoutNode.getPaddingLeft()}px`;
         this.domContainer.style.paddingRight = `${this.layoutNode.getPaddingRight()}px`;
         this.domContainer.style.lineHeight = '1em';
+        this.domContainer.style.textAlign = style.textAlign;
+        this.domContainer.style.direction = style.direction;
```

Comment:
As mentioned before in general comments, these must also be implemented in the layout engine so that conversions between screen coordinates and document position remain correct.